src/Adding Expense/AddExpense/AddExpense/AddExpense.jsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom"; 
+import axios from 'axios'; 
+import AddPayee from "../AddPayee/AddPayee";
+import "./AddExpense.css";
+
+export default function AddExpense() {
+    const { userId } = useParams();
+    const [title, setTitle] = useState("");
+    const [amount, setAmount] = useState();
+    const [isValid, setIsValid] = useState(false);
+    const [currentUser, setCurrentUser] = useState(null);
+
+    useEffect(() => {
+        if (userId) {
+            axios.get(`http://localhost:8080/Splitwise-App/users/${userId}`)
```

Comment:
You can create a generic api caller and try to re use that everywhere you are calling apis .

Like this -
```
export function createService(config) {
  return async (pathParams = {}, queryParams = {}, data = {}, headers = {}) => {
    const urlWithParams = generatePath(url, pathParams);
    const { baseUrl, url, method, headers: configHeaders } = config;
    try {
      return await axios({
        baseURL: baseUrl,
        url: urlWithParams,
        method,
        params: queryParams,
        data,
        headers: {
          ...configHeaders,
          ...headers,
          token,
        },
      });
    } catch (error) {
      if (error.response) {
        return Promise.reject(error.response.data);
      }
      return Promise.reject(error.message);
    }
  };
}
```
It's fine for now , but you can think of how can generalize the API caller as a standalone function and re use it.